services-camel/src/main/java/edu/unc/lib/cdr/routing/MetaServicesRouter.java

```diff
@@ -42,38 +42,43 @@ public void configure() throws Exception {
                 // Trigger binary processing after an asynchronously
                 .threads(enhancementThreads, enhancementThreads, "CdrEnhancementThread")
                 .delay(simple("{{cdr.enhancement.postIndexingDelay}}"))
-                .to("direct:process.creation");
-
-        from("direct:process.creation")
-            .routeId("ProcessCreationEvent")
-            .log(LoggingLevel.DEBUG, "Processing binary metadata for ${headers[org.fcrepo.jms.identifier]}")
-            .removeHeaders("CamelHttp*")
-            .to("fcrepo:{{fcrepo.baseUrl}}?preferInclude=ServerManaged&accept=text/turtle")
-            .process(mdProcessor)
-            .choice()
-                .when(simple("${headers[org.fcrepo.jms.identifier]} regex '.*original_file'"))
-                    .to("direct-vm:replication")
-                    .to("direct:process.enhancements")
-                .when(simple("${headers[org.fcrepo.jms.identifier]} regex '.*techmd_fits'"))
-                    .to("direct-vm:replication")
-                .otherwise()
-                    .log(LoggingLevel.WARN, "Cannot process binary metadata for ${headers[org.fcrepo.jms.identifier]}")
-            .end();
+                .removeHeaders("CamelHttp*")
+                .to("fcrepo:{{fcrepo.baseUrl}}?preferInclude=ServerManaged&accept=text/turtle")
+                .process(mdProcessor)
+                .multicast()
+                .to("direct:process.binary.original", "direct:process.solr");
 
+        from("direct:process.binary.original")
+            .routeId("ProcessOriginalBinary")
+            .filter(simple("${headers[org.fcrepo.jms.identifier]} regex '.*(original_file|techmd_fits)'"
+                    + " && ${headers[org.fcrepo.jms.resourceType]} contains '" + Binary.getURI() + "'"))
+                .choice()
+                    .when(simple("${headers[org.fcrepo.jms.identifier]} regex '.*original_file'"))
+                        .to("direct-vm:replication")
+                        .to("direct:process.enhancements")
+                    .when(simple("${headers[org.fcrepo.jms.identifier]} regex '.*techmd_fits'"))
+                        .to("direct-vm:replication")
+                    .otherwise()
+                        .log(LoggingLevel.WARN, "Cannot process binary metadata for ${headers[org.fcrepo.jms.identifier]}")
+                .end();
+        
         from("direct:process.enhancements")
-            .routeId("ProcessEnhancements")
+            .routeId("AddBinaryEnhancements")
             .multicast()
-                .to("direct-vm:imageEnhancements","direct-vm:extractFulltext");
+            .to("direct-vm:imageEnhancements","direct-vm:extractFulltext");
+
+        from("direct:process.solr")
+            .routeId("IngestSolrIndexing")
+            .log(LoggingLevel.DEBUG, "Ingest solr indexing for ${headers[org.fcrepo.jms.identifier]}")
+            .filter(simple("${headers[org.fcrepo.jms.resourceType]} not contains '" + Binary.getURI() + "'"))
```

Comment:
This is too permissive, and would let through DepositRecords with the current stack.  I'd suggest listing the types that are allowed rather than aren't allowed, since we are likely to add more (currently) unindexed types, such as premis events, agents, agreements, licenses, etc.  The types that should be indexed are Collections, Units, Works, FileObjects